### PR TITLE
Adding tests to ALF view 

### LINKED ISSUE #26878: ALFView unit test
ALFView needs unit tests.

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**
ALF view needs some unit tests. This PR has two sets of changes:
1. Adding interfaces to the GUI to allow for mocking
2. Adding the unit tests

**To test:**
Check that ALF view still works as expected.

Open Mantid
Go to custom interfaces -> Direct -> ALF view
An interface should appear
In the run box type 80511 and press enter
Wait a little while and an updated instrument should appear in the GUI
Go to the pick tab
Select the tube option
Change the subplot from the pick tab to integrate (right click)
Change the axis to out of plane angle (right click)
On the instrument view, right click above a bright spot and select export tube
A plot should appear on the far right of the GUI
You should be able to press the fit button and it will do a fit.

Fixes #26878 . 

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/widgets/instrumentview/test/PlotFitAnalysisPanePresenterTest.h [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  void setUp() override {
+    m_view = new NiceMock<MockPlotFitAnalysisPaneView>();
+    m_model = new MockPlotFitAnalysisPaneModel();
+    m_presenter = new PlotFitAnalysisPanePresenter(m_view, m_model);
+  }
+
+  void tearDown() override {
+    AnalysisDataService::Instance().clear();
+    delete m_view;
+    delete m_presenter;
```
Comment (alicerussell1) [inline qt/widgets/instrumentview/test/PlotFitAnalysisPanePresenterTest.h:52]: Should the model also be deleted here?

**Review by alicerussell1 — CHANGES_REQUESTED** [review]
This still works as it did before, just a couple of comments/questions below

File: qt/scientific_interfaces/Direct/test/ALFCustomInstrumentModelTest.h [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+  void test_isDataValidDSpace() {
+    auto data = mockALFData("ALF_tmp", "ALF", 6113, false);
+    std::map<std::string, bool> isDataValid = m_model->isDataValid();
+
+    TS_ASSERT(isDataValid["IsValidInstrument"]);
+    TS_ASSERT(isDataValid["IsItDSpace"]);
+  }
+
+  void test_transformData() { return; }
```
[the remark this answers is not in the document]
Comment (AnthonyLim23) [reply qt/scientific_interfaces/Direct/test/ALFCustomInstrumentModelTest.h:109]: I was going to write a test, but it just runs 2 algorithms. So nothing to really test

File: qt/widgets/instrumentview/test/PlotFitAnalysisPanePresenterTest.h (hunk 1 above)
Comment (AnthonyLim23) [reply qt/widgets/instrumentview/test/PlotFitAnalysisPanePresenterTest.h:52]: the model should be deleted as part of the presenter destructor. Adding it as `delete m_model` causes a seg fault. So I have just made it equal to `nullptr`

**Review by alicerussell1 — APPROVED** [review]
Thanks for those changes. All looks good to me 👍

**Pull request merged**